ingestion-edge/ingestion_edge/util.py

```diff
@@ -4,41 +4,102 @@
 
 """Utilities."""
 
+from google.cloud.pubsub_v1.gapic.publisher_client import PublisherClient
+from google.cloud.pubsub_v1.publisher.exceptions import PublishError
+from google.cloud.pubsub_v1.types import BatchSettings, PublishRequest, PubsubMessage
+from typing import List, Optional
 import asyncio
 
 
-def async_wrap(future, loop=None):
-    """Wrap a threading-like future in an async future that can be awaited.
+class AsyncioBatch:
+    """Batch for google.cloud.pubsub_v1.PublisherClient in asyncio.
 
-    example:
-
-    async def func():
-        future = ...
-        await wrap(future)
+    Work around for https://github.com/googleapis/google-cloud-python/issues/7104
```

Comment:
```suggestion
    Workaround for https://github.com/googleapis/google-cloud-python/issues/7104
```

Apparently, Webster's has the noun as "work-around", but the rest have it as a single unhyphenated word. I do what the OED says and they say "workaround".

This is a very important comment.